dotCMS/src/main/java/com/dotcms/rest/api/v1/apps/AppsResource.java

```diff
@@ -433,4 +434,79 @@ public final Response deleteApp(
         }
     }
 
+    /**
+     * Secrets export
+     * @param request
+     * @param response
+     * @param exportSecretForm
+     * @return
+     */
+    @POST
+    @Path("/export")
+    @JSONP
+    @NoCache
+    @Produces(MediaType.APPLICATION_OCTET_STREAM)
+    public final Response exportSecrets(
+            @Context final HttpServletRequest request,
+            @Context final HttpServletResponse response,
+            final ExportSecretForm exportSecretForm
+    ) {
+        exportSecretForm.checkValid();
+        try {
+            final InitDataObject initData =
+                    new WebResource.InitBuilder(webResource)
+                            .requiredBackendUser(true)
+                            .requiredFrontendUser(false)
+                            .requestAndResponse(request, response)
+                            .rejectWhenNoUser(true)
+                            .init();
+            final User user = initData.getUser();
+            //no need to close i'll get closed upon writing the response
+            final InputStream stream = helper.exportSecrets(exportSecretForm, user);
```

Comment:
Codacy found an issue: [Ensure that resources like this InputStream object are closed after use](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5931733)

Reply:
no need to close it'll get closed upon writing the response

Reply:
you can hide it here:

```
return Response.ok(helper.exportSecrets(exportSecretForm, user), MediaType.APPLICATION_OCTET_STREAM)
                    .header("content-disposition", "attachment; filename=appSecrets.export")
                    .build();
```